src/riak_dt_map.erl

```diff
@@ -445,6 +455,11 @@ defer_remove(Clock, Ctx, Field, Deferred) ->
 
 %% @doc merge two `map()'s.
 -spec merge(map(), map()) -> map().
+merge({_LHSC, LHSE, LHSD}=LHS, {_RHSC, RHSE, RHSD}=RHS) when is_list(LHSE);
```

Comment:
Regarding this and `equal/2` below, is there any possibility that the entries or the deferreds will be a `dict` (or list) and not the other? If not, then the guard can be simplified, because `to_v2/1` expects that both are lists.